cli/routerkeygen-cli.cpp

```diff
@@ -9,15 +9,116 @@
 
 #include "qcmdlineparser/qcmdlineparser.h"
 
+static void generateCode(QString &ssid, QString &mac)
+{
+    QScanResult wifi(ssid, mac);
+    WirelessMatcher wm;
+
+    // Add keygens
+    wifi.checkSupport(wm);
+
+    // Calculate!
+    QVector<Keygen *> * keygens = wifi.getKeygens();
+    if (keygens == nullptr)
+        return; /* Out of memory */
+
+    // Print keys
+    // Format is MAC<TAB>Password
+    for (int i = 0; i < keygens->size(); ++i) {
+        try {
+            QVector<QString> r = keygens->at(i)->getResults();
+            foreach (QString s, r) {
+                std::cout<< mac.toUtf8().data() << '\t' << s.toLatin1().data() << std::endl;
```

Comment:
Not sure \t is a good delimiter, but will do it. More important, if there is supplied -k (for keygen name), please add it in the beginning as first field.